pkg/connector/org.go

```diff
@@ -200,6 +216,115 @@ func (o *orgResourceType) Grants(
 	return rv, pageToken, reqAnnos, nil
 }
 
+func (o *orgResourceType) Grant(ctx context.Context, principal *v2.Resource, en *v2.Entitlement) (annotations.Annotations, error) {
+	l := ctxzap.Extract(ctx)
+
+	if principal.Id.ResourceType != resourceTypeUser.Id {
+		l.Error(
+			"github-connectorv2: only users can be granted org admin",
+			zap.String("principal_type", principal.Id.ResourceType),
+			zap.String("principal_id", principal.Id.Resource),
+		)
+		return nil, fmt.Errorf("github-connectorv2: only users can be granted org membership")
+	}
+
+	if en.Id != entitlement.NewEntitlementID(en.Resource, orgRoleAdmin) {
+		return nil, fmt.Errorf("github-connectorv2: invalid entitlement id: %s", en.Id)
+	}
+
+	orgName, err := getOrgName(ctx, o.client, en.Resource.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	principalID, err := strconv.ParseInt(principal.Id.Resource, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	user, _, err := o.client.Users.GetByID(ctx, principalID)
+	if err != nil {
+		return nil, fmt.Errorf("github-connectorv2: failed to get user: %w", err)
+	}
+
+	membership, _, err := o.client.Organizations.GetOrgMembership(ctx, user.GetLogin(), orgName)
+	if err != nil {
+		return nil, fmt.Errorf("github-connectorv2: failed to get org membership: %w", err)
+	}
+
+	if membership.GetRole() == orgRoleAdmin {
+		l.Debug("githubv2-connector: user is already an admin of the org")
```

Comment:
wish we could codify this behavoir via an annoation or somehting, returning `nil, nil` feels weird.